### DISCUSSION TITLE
Tablet error: "DeviceUnauthorizedAccessException: Not permitted to open HID class device at /dev/hidraw7."

### DISCUSSION THREAD

**Discussion by soopax** [ppy/osu#24891] [Q&A] [answered]
![Screenshot from 2023-09-23 15-42-40](https://github.com/ppy/osu/assets/101510334/149fe8bb-a607-498b-88c6-3305e31216f1)

The tablet works fine outside of osu! and even works in osu! without tablet support on, but trying to turn the tablet support on gives this error: "DeviceUnauthorizedAccessException: Not permitted to open HID class device at /dev/hidraw7."

* Debian 6.1.52-1 (2023-09-07) x86_64 GNU/Linux
* Wacom CTL-472-N
* Affecting AppImage, Flatpak, local build of aea7f81f1c1751858e457645c7d541ec5ee39fac

### Logs
[database.log](https://github.com/ppy/osu/files/12706469/database.log)
[input.log](https://github.com/ppy/osu/files/12706470/input.log)
[network.log](https://github.com/ppy/osu/files/12706471/network.log)
[performance.log](https://github.com/ppy/osu/files/12706472/performance.log)
[performance-audio.log](https://github.com/ppy/osu/files/12706473/performance-audio.log)
[performance-draw.log](https://github.com/ppy/osu/files/12706474/performance-draw.log)
[performance-input.log](https://github.com/ppy/osu/files/12706475/performance-input.log)
[performance-update.log](https://github.com/ppy/osu/files/12706476/performance-update.log)
[runtime.log](https://github.com/ppy/osu/files/12706477/runtime.log)

### Version
* osu! 2023.914.0-lazer

**Comment by sineplusx** [comment]
Since you said the tablet works outside of lazer: Are you using OpenTabletDriver or the (kernel module) Wacom driver? If OpenTabletDriver works with your system config, then I dont see a reason for it to not work embedded in lazer as well.

**Reply by soopax** [reply]
The kernel module was installed automatically, I haven't installed OpenTabletDriver separately. I solved the problem (see https://github.com/ppy/osu/discussions/24891#discussioncomment-7090174)

**Comment by soopax** [comment, marked as answer]
Problem solved, the problem was caused by missing udev rules. I followed the instructions at https://opentabletdriver.net/Wiki/FAQ/Linux (Failed to open device streams)

* Installed dotnet 7 using instructions from https://learn.microsoft.com/en-us/dotnet/core/install/linux-debian

```
git clone https://github.com/OpenTabletDriver/OpenTabletDriver.git
cd ./OpenTabletDriver
./generate-rules.sh
sudo mv ./bin/99-opentabletdriver.rules /etc/udev/rules.d/99-opentabletdriver.rules
```
After doing the commands and rebooting, it works now.
